app/client/src/components/pages/Community.Tabs.Monitoring.js

```diff
@@ -0,0 +1,1195 @@
+        // counts for each lower-tier characteristic group
+        stationTotalsByCategory: JSON.stringify(
+          station.properties.characteristicGroupResultCount,
+        ),
+        // counts for each top-tier characteristic group
+        stationTotalsByGroup: {},
+        // create a unique id, so we can check if the monitoring station has
+        // already been added to the display (since a monitoring station id
+        // isn't universally unique)
+        uniqueId:
+          `${station.properties.MonitoringLocationIdentifier}-` +
+          `${station.properties.ProviderName}-` +
+          `${station.properties.OrganizationIdentifier}`,
+      };
+
+      allMonitoringLocations.push(monitoringLocation);
+
+      // build up the monitoringLocationToggles and monitoringLocationGroups
+      let locationAddedToGroup = false;
+      const subGroupsAdded = [];
+
+      characteristicGroupMappings.forEach((mapping) => {
+        for (const subGroup in station.properties
+          .characteristicGroupResultCount) {
+          // if characteristic group exists in switch config object
+          if (mapping.groupNames.includes(subGroup)) {
+            subGroupsAdded.push(subGroup);
+            // if switch group (w/ label key) already exists, add the stations to it
+            if (monitoringLocationGroups[mapping.label]) {
+              monitoringLocationGroups[mapping.label].stations.push(
+                monitoringLocation,
+              );
+              // else, create the group (w/ label key) and add the station
+            } else {
+              monitoringLocationGroups[mapping.label] = {
+                label: mapping.label,
+                stations: [monitoringLocation],
+                toggled: true,
+              };
+            }
+            locationAddedToGroup = true;
+            // add the lower-tier group counts to the corresponding top-tier group counts
+            if (
+              !monitoringLocation.stationTotalsByGroup.hasOwnProperty(
+                mapping.label,
+              )
+            ) {
+              monitoringLocation.stationTotalsByGroup[mapping.label] =
+                station.properties.characteristicGroupResultCount[subGroup];
+            } else {
+              monitoringLocation.stationTotalsByGroup[mapping.label] +=
+                station.properties.characteristicGroupResultCount[subGroup];
+            }
+          }
+        }
+      });
+
+      // add any leftover lower-tier group counts to the 'Other' top-tier group
+      if (!monitoringLocation.stationTotalsByGroup['Other']) {
+        monitoringLocation.stationTotalsByGroup['Other'] = 0;
+      }
+      for (const subGroup in station.properties
+        .characteristicGroupResultCount) {
+        if (!subGroupsAdded.includes(subGroup)) {
+          monitoringLocation.stationTotalsByGroup['Other'] +=
+            station.properties.characteristicGroupResultCount[subGroup];
+        }
+      }
```

Comment:
```suggestion
      for (const subGroup in station.properties
        .characteristicGroupResultCount) {
        if (!subGroupsAdded.includes(subGroup)) {
          monitoringLocation.stationTotalsByGroup['Other'] +=
            station.properties.characteristicGroupResultCount[subGroup];

          if (
            !monitoringLocationGroups['Other'].characteristicGroups.includes(
              subGroup,
            )
          ) {
            monitoringLocationGroups['Other'].characteristicGroups.push(
              subGroup,
            );
          }
        }
      }
```

Sorry for not seeing this earlier. 

I noticed another issue. If you only select `Other` the download will download everything. To work around this we need to dynamically build up all of the characteristic groups that get lumped into the other category. This suggestion should work.

You will also need to update the code on lines 784 - 787 to be like the following: 
```
let monitoringLocationGroups: MonitoringLocationGroups = {
      All: { label: 'All', stations: [], toggled: true },
      Other: { label: 'Other', stations: [], toggled: true,  characteristicGroups: [] },
};
```

You will also need to update the `buildFilter` function to use the characteristic groups from this new array if the selected named equals `Other`. Should look something like:

```
      for (const name of selectedNames) {
          if (name === 'Other') {
            filter +=
              '&characteristicType=' +
              monitoringGroups[name].characteristicGroups.join(
                '&characteristicType=',
              );
          } else {
            filter +=
              '&characteristicType=' +
              groups[name].join('&characteristicType=');
          }
      }
```

Reply:
Good call, yeah that's what I was trying to fix by adding the missing group names to `characteristicGroupMappings`, but doing it dynamically is a lot cleaner.